### PR TITLE
Frog Basic Mob Refactor

### PR DESCRIPTION
**Pull request by Comxy**
## About The Pull Request
Refactors the frog into a basic mob. The frog now does the same as the old frog and can now properly be commanded by the regal rats.
## Changelog
:cl:
refactor: Refractors the frog into a basic mob
/:cl:

### CODE REVIEW

File: code/datums/ai/basic_mobs/targetting_datums/basic_targetting_datum.dm [hunk 1]
```diff
@@ -15,6 +15,8 @@
 /datum/targetting_datum/basic
 	/// When we do our basic faction check, do we look for exact faction matches?
 	var/check_factions_exactly = FALSE
+	/// Only hostile against mobs when damaged?
+	var/check_hostile = FALSE
```
Comment (Jacquerel) [inline code/datums/ai/basic_mobs/targetting_datums/basic_targetting_datum.dm:19]: I would use the `/datum/element/ai_retaliate` element and `/datum/ai_planning_subtree/target_retaliate` subtree for this, see the implementation on Pigs.

File: code/modules/mob/living/basic/vermin/frog.dm [hunk 2]
```diff
@@ -32,13 +34,20 @@
...
+	ai_controller = /datum/ai_controller/basic_controller/frog
+
 	var/stepped_sound = 'sound/effects/huuu.ogg'
 	///How much of a reagent the mob injects on attack
 	var/poison_per_bite = 3
 	///What reagent the mob injects targets with
 	var/poison_type = /datum/reagent/drug/space_drugs
 
-/mob/living/simple_animal/hostile/retaliate/frog/Initialize(mapload)
+	var/frog_cell_line = CELL_LINE_TABLE_COCKROACH
```
Comment (Jacquerel) [inline code/modules/mob/living/basic/vermin/frog.dm:48]: Is this right?

File: code/modules/mob/living/simple_animal/hostile/regalrat.dm [hunk 3]
```diff
@@ -314,7 +314,7 @@
 /// Turns a mouse into an angry mouse
 /datum/action/cooldown/riot/proc/convert_mouse(mob/living/basic/mouse/nearby_mouse)
 	// This mouse is already rat controlled, let's not bother with it.
-	if (istype(nearby_mouse.ai_controller, /datum/ai_controller/basic_controller/mouse/rat))
+	if (istype(nearby_mouse.ai_controller, /datum/ai_controller/basic_controller/mouse/rat) || nearby_mouse.stat == DEAD)
```
Comment (Jacquerel) [inline code/modules/mob/living/simple_animal/hostile/regalrat.dm:317]: Mice and roaches delete on death, is this necessary for those or only for frogs?

**Comment by Jacquerel** [comment]
I for one am excited about the potential of commanding frogs to attack people and giving as many people as possible the Tripping screen filter.

File: code/modules/mob/living/basic/vermin/frog.dm (hunk 2 above)
Comment (Comxy) [reply code/modules/mob/living/basic/vermin/frog.dm:48]: Oh god I forgot

File: code/datums/ai/basic_mobs/targetting_datums/basic_targetting_datum.dm (hunk 1 above)
Comment (tralezab) [reply code/datums/ai/basic_mobs/targetting_datums/basic_targetting_datum.dm:19]: I agree with Jacquerel.

File: code/datums/ai/basic_mobs/targetting_datums/basic_targetting_datum.dm [hunk 4]
```diff
@@ -35,6 +37,9 @@
 		var/mob/living/L = the_target
 		if(faction_check(living_mob, L) || L.stat)
 			return FALSE
+		else if(check_hostile)
+			if(living_mob.maxHealth == living_mob.health)
+				return FALSE
```
Comment (tralezab) [inline code/datums/ai/basic_mobs/targetting_datums/basic_targetting_datum.dm:42]: This was awful behavior of it becoming hostile to everything on old retaliate mobs, fuck that

File: code/modules/mob/living/basic/vermin/frog.dm (hunk 2 above)
Comment (tralezab) [reply code/modules/mob/living/basic/vermin/frog.dm:48]: 🤭

File: code/modules/mob/living/basic/vermin/frog.dm [hunk 5]
```diff
@@ -57,15 +66,41 @@
 	)
 	AddElement(/datum/element/connect_loc, loc_connections)
 	AddElement(/datum/element/venomous, poison_type, poison_per_bite)
-	add_cell_sample()
+	AddElement(/datum/element/swabable, frog_cell_line, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
```
Comment (tralezab) [inline code/modules/mob/living/basic/vermin/frog.dm:69]: use the variable you declared? does it even make sense as a variable? probably not

File: code/modules/mob/living/basic/vermin/frog.dm [hunk 6]
```diff
@@ -57,15 +66,41 @@
...
+	idle_behavior = /datum/idle_behavior/idle_random_walk
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/pet_planning,
+		/datum/ai_planning_subtree/random_speech/frog,
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree/frog,
+	)
+
+/datum/targetting_datum/basic/frog
+	check_hostile = TRUE
```
Comment (tralezab) [inline code/modules/mob/living/basic/vermin/frog.dm:94]: you don't use this and as per the other review, it's not a good idea.

**Review by tralezab — CHANGES_REQUESTED** [review]
you need to make an entry into the update paths tool, as well.

**Comment by san7890** [comment]
> you need to make an entry into the update paths tool, as well.

https://github.com/tgstation/tgstation/tree/master/tools/UpdatePaths

**Comment by Comxy** [comment]
Like that?

**Comment by san7890** [comment]
no, that'll do the opposite. you need what the old path was on the left, and the new path on the right.

**Comment by calliophis-maculiceps** [comment]
lets us tame frogs PLEASE

**Pull request merged**